tests/bgp/test_bgp_speaker.py

```diff
@@ -324,29 +459,107 @@ def bgp_speaker_announce_routes_common(common_setup_teardown,
                 log_file="/tmp/bgp_speaker_test.FibTest.log",
                 socket_recv_size=16384)
 
+    #To verify that AS_PATH has valid ASN for prefix 
+    if family == "v4":
+        bgp_table_output = duthost.shell("vtysh -c \"show ip bgp {} longer-prefixes\" -c \"exit\"".format(prefix))
+        assert len(bgp_table_output) >= 5
+        for values in bgp_table_output["stdout_lines"]:
+            if prefix in values:
+                if nexthop_ips[1].ip and nexthop_ips[2].ip and neighbor_asn in values:
+                    logger.info("Bgp AS Path for {} is {}".format(prefix,values))
+                    assert "true"
+            if peer_range in values:
+                if nexthop_ips[0].ip and neighbor_asn in values:
+                    logger.info("Bgp AS Path for {} is {}".format(peer_range,values))
+                    assert "true"
+    elif family == "v6":
+        bgp_table_output = duthost.shell("vtysh -c \"show ip bgp ipv6\" -c \"exit\"")
+        for values in bgp_table_output["stdout_lines"]:
+            if prefix in values:
+                if bgp_dut_asn and neighbor_asn in values:
+                    logger.info("Bgp AS Path for {} is {}".format(prefix,values))
+                    assert "true"
+
     logger.info("Withdraw routes")
     withdraw_route(ptfip, lo_addr, prefix, nexthop_ips[1].ip, port_num[0])
     withdraw_route(ptfip, lo_addr, prefix, nexthop_ips[2].ip, port_num[1])
     withdraw_route(ptfip, lo_addr, peer_range, vlan_ips[0].ip, port_num[2])
 
-    logger.info("Nexthop ip%s tests are done" % family)
+    logger.info("Nexthop ip%s tests are done for test %s" % (family, Asntype))
 
-
-@pytest.mark.parametrize("ipv4, ipv6, mtu", [pytest.param(True, False, 1514)])
-def test_bgp_speaker_announce_routes(common_setup_teardown, tbinfo, duthosts, rand_one_dut_hostname, ptfhost, ipv4, ipv6, mtu, vlan_mac):
+@pytest.mark.parametrize('common_setup_teardown', [
+      ['']
```

Comment:
Is this for the existing test scenario? Is the 2 byte scenario same as the existing scenario? Then why add different test cases?

Reply:
There are 3 test case for existing test case. I added 4. 2 for 2 byte(ASN 2 byte and bgp speaker 4 byte) and 2  for 4byte ASN(both DUT and bgp speaker 4 byte ASN)